src/main/java/seedu/address/model/gameentry/DatePlayed.java

```diff
@@ -22,33 +27,61 @@ public DatePlayed() {
     }
 
     /**
-     * Constructs DatePlayed object.
+     * Constructs DatePlayed object from a String.
      *
-     * @param date Date that this object represents.
-     * @param isTimeIndicated boolean to indicate whether to ignore minute timestamp.
+     * @param datePlayedString String representing DatePlayed.
      */
-    public DatePlayed(Date date, boolean isTimeIndicated) {
-        this.datePlayed = date;
-        this.isTimeIndicated = isTimeIndicated;
-    }
+    public DatePlayed(String datePlayedString) throws InvalidDateFormatException {
+        requireNonNull(datePlayedString);
+        if (!isValidDatePlayedString(datePlayedString)) {
+            throw new InvalidDateFormatException();
+        }
+        String trimmedString = datePlayedString.trim();
+        Date date;
+        try {
```

Comment:
I think we can use a format checking method to decide on which parser to use instead of using ParseException as the control flow.

Reply:
Not crucial, can cover during refactoring.